src/patchelf.cc

```diff
@@ -1082,6 +1073,16 @@ std::string ElfFile<ElfFileParamNames>::getInterpreter()
 template<ElfFileParams>
 void ElfFile<ElfFileParamNames>::modifySoname(sonameMode op, const std::string & newSoname)
 {
+
+    bool soNameModified = false;
+
+    Finally finally([&]() {
```

Comment:
I would actually prefer the old code in cases where we don't need to handle early returns
to avoid hidden control flow. This one is such a case.